app/lib/features/profile/pages/my_profile_page.dart

```diff
@@ -1,121 +1,42 @@
-
-  @override
-  void initState() {
-    super.initState();
-    newUsername.text = widget.account.profile.displayName ?? '';
-  }
-
-  @override
-  Widget build(BuildContext context) {
-    final account = widget.account;
-    return AlertDialog(
-      title: const Text('Change your display name'),
-      content: Form(
-        key: _formKey,
-        child: Column(
-          mainAxisSize: MainAxisSize.min,
-          children: [
-            Padding(
-              padding: const EdgeInsets.all(5),
-              child: TextFormField(controller: newUsername),
-            ),
-          ],
-        ),
-      ),
-      actions: <Widget>[
-        TextButton(
-          onPressed: () => Navigator.pop(context, null),
-          child: const Text('Cancel'),
-        ),
-        TextButton(
-          onPressed: () {
-            if (_formKey.currentState!.validate()) {
-              final currentUserName = account.profile.displayName;
-              final newDisplayName = newUsername.text;
-              if (currentUserName != newDisplayName) {
-                Navigator.pop(context, newDisplayName);
-              } else {
-                Navigator.pop(context, null);
-              }
-              return;
-            }
-          },
-          child: const Text('Submit'),
-        ),
-      ],
-    );
-  }
-}
-
-class MyProfilePage extends ConsumerWidget {
-  static const logoutKey = Key('my-profile-logout');
+class MyProfilePage extends StatelessWidget {
   static const displayNameKey = Key('my-profile-display-name');
-  static const settingsKey = Key('my-profile-settings');
 
   const MyProfilePage({super.key});
 
   Future<void> updateDisplayName(
     AccountProfile profile,
+    String displayName,
     BuildContext context,
     WidgetRef ref,
   ) async {
-    final TextEditingController newName = TextEditingController();
-    newName.text = profile.profile.displayName ?? '';
-
-    final newText = await showDialog<String>(
-      context: context,
-      builder: (BuildContext context) => ChangeDisplayName(account: profile),
-    );
-
-    if (newText != null && context.mounted) {
-      showAdaptiveDialog(
-        context: context,
-        builder: (context) => DefaultDialog(
-          title: Text(
-            'Updating Display Name',
-            style: Theme.of(context).textTheme.titleSmall,
-          ),
-          isLoader: true,
-        ),
-      );
-      await profile.account.setDisplayName(newText);
+    if (displayName.isNotEmpty && context.mounted) {
+      EasyLoading.show(status: 'Updating profile data');
+      await profile.account.setDisplayName(displayName);
       ref.invalidate(accountProfileProvider);
 
       if (!context.mounted) {
         return;
       }
-      Navigator.of(context, rootNavigator: true).pop();
-      customMsgSnackbar(context, 'Display Name update submitted');
+
+      // close loading
+      EasyLoading.dismiss();
+
+      context.pop();
```

Comment:
Where does this pop to? Back to settings menu? If that is, we don't want to navigate back. Not sure if its needed since Loader is changed and dismissed already by here.

Reply:
Yes, it is for going back to setting menu. Open for suggestion here :)